lib/perlfaq4.pod

```diff
@@ -273,32 +273,63 @@ are left as an exercise to the inclined reader.
 
 =head2 Why doesn't & work the way I want it to?
 
-The behavior of binary arithmetic operators depends on whether they're
-used on numbers or strings. The operators treat a string as a series
-of bits and work with that (the string C<"3"> is the bit pattern
-C<00110011>). The operators work with the binary form of a number
-(the number C<3> is treated as the bit pattern C<00000011>).
-
-So, saying C<11 & 3> performs the "and" operation on numbers (yielding
-C<3>). Saying C<"11" & "3"> performs the "and" operation on strings
-(yielding C<"1">).
-
-Most problems with C<&> and C<|> arise because the programmer thinks
-they have a number but really it's a string or vice versa. To avoid this,
-stringify the arguments explicitly (using C<""> or C<qq()>) or convert them
-to numbers explicitly (using C<0+$arg>). The rest arise because
-the programmer says:
-
-    if ("\020\020" & "\101\101") {
-        # ...
-    }
+Perl's C<&> bitwise operator works on both numbers and strings,
+sometimes producing surprising results when you expected a number
+but received a string. You probably expected perl to automatically
+convert the operands to numbers like the mathematical operators would.
+Instead, perl treats string operands as a bitvectors.
```

Comment:
```suggestion
Instead, perl treats string operands as bitvectors.
```

Reply:
fixed in ddd936a77b5b6f4aa41024f68d7b2e1013ecfaec (which isn't showing up yet)